### PR TITLE
[build] Update GHCI

### PR DESCRIPTION
**Pull request by sushantmane**
## Update GHCI
Split CI into three parts:
1) StaticAnalysisAndUnitTests - workflow for running static analysis and unit tests
    - Static analysis will be run with JDK17
2) E2ETests -  workflow for running end to end tests
3) CompatibilityTests - workflow for running compatibility tests: Avro, Alpini unit,
  Alpini functional tests, and Pulsar Venice integration tests

| Job                          |             JDK    |
| ------------------- |:-------------:|
| Unit Tests                | 8, 11, & 17      |
| Static Analysis        | 17                   |
| E2E Tests                |  17                  |
| Avro Comatibility Tests | 17 |
| Alpini Tests  | 17 |
| PulsarVeniceIntegrationTests | 17 |

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: .github/workflows/VeniceCI-StaticAnalysisAndUnitTests.yml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+          tar -zcvf ${{ github.job }}-jdk${{ matrix.jdk }}-logs.tar.gz ${{ github.job }}-artifacts
+      - name: Upload Build Artifacts
+        if: success() || failure()
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ github.job }}
+          path: ${{ github.job }}-jdk${{ matrix.jdk }}-logs.tar.gz
+          retention-days: 30
+
+  SacUtCompletion:
```
Comment (adamxchen) [inline .github/workflows/VeniceCI-StaticAnalysisAndUnitTests.yml:103]: nit: can we avoid using confusing abbreviation names for the workflow? `E2ECompletion` sounds fine, but `SacUtCompletion` and `CompatTestsCompletion` aren't very self-explanatory at first glance.

Comment (sushantmane) [reply .github/workflows/VeniceCI-StaticAnalysisAndUnitTests.yml:103]: Good point. I'll replace `SacUtCompletion` with `StaticAnalysisAndUnitTestsCompletion` and `CompatTestsCompletion` with `CompatibilityTestsCompletion`. Does that sound better?

Comment (adamxchen) [reply .github/workflows/VeniceCI-StaticAnalysisAndUnitTests.yml:103]: Yes, please!

Comment (sushantmane) [reply .github/workflows/VeniceCI-StaticAnalysisAndUnitTests.yml]: Done. Thanks!

File: .github/workflows/VeniceCI-CompatibilityTests.yml [hunk 2]
```diff
@@ -0,0 +1,264 @@
+# GitHub Actions workflow for running compatibility tests: Avro, Alpini unit, Alpini functional tests, and Pulsar Venice integration tests
+
+name:  TestsDeCompatibilite
```
Comment (FelixGV) [inline .github/workflows/VeniceCI-CompatibilityTests.yml:3]: A little bit of French sprinkled in there, why not 😁

File: .github/workflows/VeniceCI-CompatibilityTests.yml [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+        with:
+          fetch-depth: 0
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        if: steps.check_alpini_files_changed.outputs.alpini == 'true'
+        with:
+          java-version: ${{ matrix.jdk }}
+          distribution: 'temurin'
+          cache: 'gradle'
+      # - name: Allow Deprecated TLS versions for Alpini tests
```
Comment (FelixGV) [inline .github/workflows/VeniceCI-CompatibilityTests.yml:83]: Is this commented stuff needed?

**Review by FelixGV — APPROVED** [review]
Thanks a lot! LGTM. Seems like we'll need to tweak the branch protection rules for the checks to pass though 🤔

File: .github/workflows/VeniceCI-CompatibilityTests.yml (hunk 3 above)
Comment (nisargthakkar) [reply .github/workflows/VeniceCI-CompatibilityTests.yml:83]: https://github.com/linkedin/venice/pull/244#discussion_r1132975861 - Same comment 😅

When I read through my reply, it doesn't make sense anymore as to how keeping this commented helps. It's okay to get rid of it. We can get it from git history now

**Comment by adamxchen** [comment]
> Seems like we'll need to tweak the branch protection rules for the checks to pass though

We can probably merge this by bypassing the branch protections one time for this PR and then update the branch protection rules after merging this.

File: .github/workflows/VeniceCI-CompatibilityTests.yml (hunk 3 above)
Comment (nisargthakkar) [reply .github/workflows/VeniceCI-CompatibilityTests.yml:83]: I see what I now meant in the thread. We have these two tests disabled:
```
@Test(enabled = false)
public void testLegacyClientTLS() throws Throwable {
...
}

@Test(enabled = false)
public void testLegacyServerTLS() throws Throwable {
...
}
```

**Comment by FelixGV** [comment]
> > Seems like we'll need to tweak the branch protection rules for the checks to pass though
> We can probably merge this by bypassing the branch protections one time for this PR and then update the branch protection rules after merging this.

I have changed the protections now. This is cleanly mergeable (and everything else will need to rebase).

File: .github/workflows/VeniceCI-CompatibilityTests.yml (hunk 3 above)
Comment (FelixGV) [reply .github/workflows/VeniceCI-CompatibilityTests.yml:83]: Let's take up this fight another day 😅

**Review by FelixGV — APPROVED** [review]
LFG!!!

**Comment by sushantmane** [comment]
Thanks a lot, @FelixGV , @adamxchen, and @nisargthakkar for the review!

File: .github/workflows/VeniceCI-CompatibilityTests.yml (hunk 2 above)
Comment (sushantmane) [reply .github/workflows/VeniceCI-CompatibilityTests.yml:3]: haha. It helps to have tests in nice order under checks tab: unit --> e2e --> compatibility. Plus it sounds cool 😂

**Pull request merged**